packages/infra/src/handlers/framework/context.ts

```diff
@@ -12,3 +13,32 @@ export const getContext = (): Context => storage.getStore() ?? {};
 export const getAbortSignal = () => storage.getStore()?.abortSignal;
 
 export const getRequestId = () => storage.getStore()?.requestId;
+
+export const withTimeout = async <T>(
+  task: () => Promise<T>,
+  timeoutMs: number,
+): Promise<T> => {
+  const cancelTask = new AbortController();
+  const cancelTimeout = new AbortController();
+
+  const runTask = async (): Promise<T> => {
+    try {
+      return await storage.run(
+        { ...getContext(), abortSignal: cancelTask.signal },
+        task,
+      );
+    } finally {
+      cancelTimeout.abort();
+    }
+  };
+
+  const runTimeout = async (): Promise<never> => {
```

Comment:
Can we use https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static?